uk.co.compucorp.civicrm.tasksassignments/tasksassignments.php

```diff
@@ -12,6 +16,28 @@ function tasksassignments_civicrm_config(&$config) {
 }
 
 /**
+ * @param array $wrappers
+ * @param array $apiRequest
+ */
+function tasksassignments_civicrm_apiWrappers(&$wrappers, $apiRequest) {
+  $requestJson = ArrayHelper::value('json', $_REQUEST);
+  $params = ArrayHelper::value('params', $apiRequest, []);
+  $optionGroup = ArrayHelper::value('option_group_id', $params);
+  $isActivityType = $optionGroup === 'activity_type';
+  $entity = ArrayHelper::value('entity', $apiRequest);
+  $action = ArrayHelper::value('action', $apiRequest);
+
+  $jsonFromCaseEdit = '"actTypes":["OptionValue","get",{"option_group_id":"activity_type","options":{"sort":"name","limit":0}}]';
+  $isAssignmentEdit = strpos($requestJson, $jsonFromCaseEdit) !== FALSE;
+
+  if ($isAssignmentEdit && $entity === 'OptionValue' && $action === 'get' && $isActivityType) {
+    $wrappers[] = new StripInvalidTaskTypesWrapper();
+  } elseif ($entity === 'CaseType' && $action === 'create') {
```

Comment:
since you only show activity types that belongs to civitask and cividoc components , I think there is no need to  do this check when creating caseType .. so I think you can remove this condition and the whole **AssignmentDefinitionValidationWrapper** class .